src/app/pages/credentials/new-users/all-users/user-list/user-list.component.ts

```diff
@@ -31,111 +34,102 @@ import { UsersStore } from 'app/pages/credentials/new-users/store/users.store';
   standalone: true,
   imports: [
     TranslateModule,
-    EmptyComponent,
-    UserRowComponent,
-    SearchInput1Component,
-    MatCheckbox,
-    FakeProgressBarComponent,
+    AsyncPipe,
+    IxTableBodyComponent,
+    IxTableComponent,
+    IxTableEmptyDirective,
+    IxTableHeadComponent,
+    IxTablePagerComponent,
+    NgTemplateOutlet,
+    UsersSearchComponent,
   ],
 })
 export class UserListComponent {
   readonly userName = injectParams('id');
-  readonly isMobileView = input<boolean>();
-  readonly toggleShowMobileDetails = output<boolean>();
-  protected readonly searchQuery = signal<string>('');
-  protected readonly window = inject<Window>(WINDOW);
-  protected readonly selection = new SelectionModel<string>(true, []);
-  protected readonly users = this.usersStore.users;
-  protected readonly isLoading = this.usersStore.isLoading;
-  protected readonly selectedUser = this.usersStore.selectedUser;
-
-  readonly isSelectedUserVisible = computed(() => {
-    return this.filteredUsers()?.some((user) => user.id === this.selectedUser()?.id);
+  readonly searchQuery = injectParams<SearchQuery<User>>((params) => {
+    return params.searchQuery;
   });
 
-  protected readonly filteredUsers = computed(() => {
-    return (this.users() || []).filter((user) => {
-      return user?.username?.toLocaleLowerCase().includes(this.searchQuery().toLocaleLowerCase());
-    });
-  });
+  readonly isMobileView = input<boolean>();
+  readonly toggleShowMobileDetails = output<boolean>();
+  readonly userSelected = output<User>();
+  // TODO: NAS-135333 - Handle case after url linking is implemented to decide when no to show selected user
+  readonly isSelectedUserVisible$ = of(true);
+  readonly dataProvider = input.required<ApiDataProvider<'user.query'>>();
 
-  protected readonly emptyConfig = computed<EmptyConfig>(() => {
-    if (this.searchQuery()?.length && !this.filteredUsers()?.length) {
-      return {
-        type: EmptyType.NoSearchResults,
```

Comment:
![image](https://github.com/user-attachments/assets/fdce2370-87b9-4953-acf4-dd02c8ab4d83)

Let's add back empty config

Reply:
Empty config is already implemented with combination of `ApiDataProvider` and `IxTableComponent`